code/go/0chain.net/sharder/blockstore/fs_store.go

```diff
@@ -2,275 +2,245 @@ package blockstore
-		err = filepath.Walk(fbs.RootDirectory,
-			func(path string, fi os.FileInfo, err error) error {
-				if err != nil {
-					return err
-				}
-				if !fi.IsDir() {
-					return nil
-				}
-				path = filepath.Join(path, s1, s2, s3, tail) // block path
-				fi, err = os.Stat(path)
-				if err != nil {
-					if os.IsNotExist(err) {
-						// can't use errors.Is(err, os.ErrNotExist) with go1.12
-						return nil // not an error (continue)
-					}
-					return err // filesystem error
-				}
-				// got the file
-				if b, err = fbs.read(hash, round); err != nil {
-					return err
-				}
-				return io.EOF // ok (just stop walking loop)
-			})
-
-		if err != io.EOF {
-			return // unexpected error
-		}
-
-		err = nil // reset the io.EOF
-
-		// err is not nil doesn't mean we have the block
-
-		if b == nil {
-			return nil, os.ErrNotExist
-		}
-
-		return // got it
-
-	*/
+func (bStore *BlockStore) Read(hash string) (*block.Block, error) {
+	return bStore.read(bStore, hash)
 }
 
-func (fbs *FSBlockStore) read(hash string, round int64) (*block.Block, error) {
-	if len(hash) != 64 {
-		return nil, encryption.ErrInvalidHash
-	}
-	fileName := fbs.getFileName(hash, round)
-	f, err := os.Open(fileName)
+func (bStore *BlockStore) readFromDisk(hash string) (*block.Block, error) {
+	bPath := filepath.Join(bStore.basePath, getBlockFilePath(hash))
+	f, err := os.Open(bPath)
 	if err != nil {
-		if os.IsNotExist(err) {
-			if viper.GetBool("minio.enabled") {
-				err = fbs.DownloadFromCloud(hash, round)
-				if err != nil {
-					return nil, err
-				}
-			}
-			f, err = os.Open(fileName)
-			if err != nil {
-				return nil, err
-
-			}
-		} else {
-			return nil, err
-		}
+		return nil, err
 	}
 	defer f.Close()
+
 	r, err := zlib.NewReader(f)
 	if err != nil {
 		return nil, err
 	}
 	defer r.Close()
-	b := fbs.blockMetadataProvider.Instance().(*block.Block)
+	b := bStore.blockMetadataProvider.Instance().(*block.Block)
 	err = datastore.ReadMsgpack(r, b)
 	if err != nil {
 		return nil, err
 	}
 	return b, nil
 }
 
-// Delete - delete from the hash of the block
-func (fbs *FSBlockStore) Delete(hash string) error {
-	return common.NewError("interface_not_implemented", "FSBlockStore cannote provide this interface")
+// ReadWithBlockSummary - read the block given the block summary
+func (bStore *BlockStore) ReadWithBlockSummary(bs *block.BlockSummary) (*block.Block, error) {
+	return bStore.read(bStore, bs.Hash)
 }
 
-// DeleteBlock - delete the given block from the file system
-func (fbs *FSBlockStore) DeleteBlock(b *block.Block) error {
-	fileName := fbs.getFileName(b.Hash, b.Round)
-	err := os.Remove(fileName)
-	if err != nil {
-		return err
+// Init checks for minimum disk size, inodes requirement and assigns
+// block storer to a variable. If any error occurs during initialization
+// it will panic.
+// It will register read and write function based on whether cache config is provided or not.
+func Init(ctx context.Context, sViper *viper.Viper) {
```

Comment:
ctx is not used

Reply:
Removing it. Context is not required as well.